Implement AOS frame support

kmarwah opened the pull request:
Add support for AOS frames, including the requested changes from @aywaldron

aywaldron commented on ait/dsn/sle/common.py:
```diff
@@ -97,6 +97,8 @@ class SLE(object):
     def __init__(self, *args, **kwargs):
         ''''''
+        self. _downlink_frame_type = ait.config.get('dsn.sle.downlink_frame_type',
+                                      kwargs.get('telemetryframetype', 'TMTransFrame'))
```
The kwargs key to check for here would be `downlink_frame_type` as well, not `telemetryframetype`.

aywaldron commented on ait/dsn/bin/ait_sle_bridge.py:
```diff
@@ -65,16 +65,17 @@ def process_pdu(raf_mngr):
 if __name__ == '__main__':
-    raf_mngr = ait.dsn.sle.RAF(hostname='atb-ocio-sspsim.jpl.nasa.gov', port=5100,
+    #put in if statement for each IP address
+    raf_mngr = ait.dsn.sle.RAF(hostname='137.228.89.105', port=5214, #ip address subject to change
```
If this hostname, port, and inst_id is project-specific information it should not be included in the PR.

kmarwah replied:
The original hostname (atb-ocio-sspsim.jpl.nasa.gov) is also project specific. Should I add a sle_hostname field in config.yaml?

aywaldron commented on ait/dsn/bin/examples/cltu_api_test.py:
```diff
@@ -38,9 +38,9 @@
+    hostname='127.0.0.1', #test for now
     port=5100,
-    inst_id='sagr=LSE-SSC.spack=Test.fsl-fg=1.cltu=cltu1',
+    inst_id='sagr=39.spack=default.fsl-fg=1.cltu=cltu1',
```
Same comment as above.

kmarwah replied:
the original "LSE-SSC" is also project specific, but from another project. Should I add another field in config.yaml for CLTU_inst_id?

kmarwah commented:
@aywaldron I removed all the project specific information I could, including some references to SSPSim that are currently in the NASA-AMMOS:master branch. I also added a few more fields in config.yaml, which means that less Python code has to be edited when instantiating a CLTU/RAF/RCF class. The only thing that I didn't include in config.yaml was inst_id, because this value changes depending on whether it's for FCLTU, RAF-online, RCF-offline, etc.

aywaldron commented on ait/dsn/bin/ait_sle_bridge.py:
```diff
@@ -65,16 +65,16 @@ def process_pdu(raf_mngr):
-                             auth_level="bind",
-                             inst_id="sagr=LSE-SSC.spack=Test.rsl-fg=1.raf=onlc1")
+    raf_mngr = ait.dsn.sle.RAF(hostname=self._hostname, port=self._port,
+                             auth_level=self._auth_level,
```
This code is not in a class so `self.` isn't going to work.

kmarwah replied:
Good catch. I removed all the redundant arguments which are already present in config.yaml and inherited from the SLE base class defined in common.py (hostname, port, auth_level). The only required argument when instantiating a RAF/RCF/CLTU class is inst_id, which isn't defined in config.yaml because it's specific to the object being instantiated.

aywaldron commented on ait/dsn/bin/examples/cltu_api_test.py:
```diff
@@ -38,10 +38,10 @@
-    inst_id='sagr=LSE-SSC.spack=Test.fsl-fg=1.cltu=cltu1',
-    auth_level="bind"
+    hostname=self._hostname,
+    port=self._port,
```
Same comment as above.

aywaldron commented:
Hi @kmarwah, thanks for updating again. Looking at the AOS transfer frame spec at https://public.ccsds.org/Pubs/732x0b3e1.pdf, page 4-2, there are a few discrepancies between your implementation and the spec. For example, in the spec the signaling field contains 4 distinct values (it is not unused), and the last 2 octets are the optional "frame header error control", not the first header pointer. Which document are you referencing in your implementation? If your implementation has some project-specific aspects to it, I will want to edit it to make it more general after merging your PR.

kmarwah commented:
hey @aywaldron, I referenced the Iris radio's ICD when building the class, so the implementation is specific to LunaH-Map and other projects using Iris (Icecube).

![image](https://user-images.githubusercontent.com/3537819/60462634-b019f380-9bfe-11e9-8d4e-12693770d1b0.png)

aywaldron commented:
@kmarwah Ok, thanks for including that reference here. I will merge this PR and leave the ticket open for us to implement the specs in the CCSDS documentation rather than the Iris ICD.
